spotify_player/src/cli/client.rs

```diff
@@ -350,3 +365,453 @@ async fn handle_playback_request(
+
+                let id = TrackId::from_id(id_s.to_owned())?;
+                let track = PlaylistData { id, name };
+
+                old_ids.insert(track);
+            }
+
+            let mut result = String::new();
+
+            // Add all new tracks
+            let mut track_buff = Vec::new();
+            let new_tracks = new_ids.difference(&old_ids);
+            let mut new_tracks_count = 0;
+            for track in new_tracks {
+                if !to_hash.contains(&track.id) {
+                    track_buff.push(PlayableId::Track(track.id.to_owned()));
+                    new_tracks_count += 1;
+
+                    if track_buff.len() > 90 {
+                        client
+                            .spotify
+                            .playlist_add_items(import_to.to_owned(), track_buff, None)
+                            .await?;
+                        track_buff = Vec::new();
+                    }
+                }
+            }
+
+            if !track_buff.is_empty() {
+                client
+                    .spotify
+                    .playlist_add_items(import_to.to_owned(), track_buff, None)
+                    .await?;
+            }
+
+            result.push_str(
+                format!(
+                    "Updated the import '{}' for '{}'.\nAdded '{}' new songs.\n",
+                    from.name, to.name, new_tracks_count
+                )
+                .as_str(),
+            );
+
+            // This is meant to prompt for deletion,
+            // May not be able to do input/output because of
+            // client and cli structure
+            let mut delete_buff = Vec::new();
+            let mut have_removed = false;
+            let deleted_tracks = old_ids.difference(&new_ids);
+            for track in deleted_tracks {
+                if !have_removed {
+                    result.push_str("The import has deleted these tracks:\n");
+                    have_removed = true;
+                }
+
+                result.push_str(format!("{}:{}\n", track.id.id(), track.name).as_str());
+                if delete {
+                    delete_buff.push(PlayableId::Track(track.id.to_owned()));
+                    if delete_buff.len() > 90 {
+                        client
+                            .spotify
+                            .playlist_remove_all_occurrences_of_items(
+                                to.id.to_owned(),
+                                delete_buff,
+                                None,
+                            )
+                            .await?;
+                        delete_buff = Vec::new();
+                    }
+                }
+            }
+
+            if delete {
+                result.push_str("These tracks have been deleted from the playlist.");
+                client
+                    .spotify
+                    .playlist_remove_all_occurrences_of_items(to.id, delete_buff, None)
+                    .await?;
+            }
```

Comment:
I see, so the `delete` option is to remove **all** tracks from the previous version of the playlist but not from the current version. Should make this more clear in the command's document instead of

```
.help("Will delete any tracks that the import does."))
.after_help("Copies all tracks into a playlist. If imported again, it adds new tracks and outputs information on deleted tracks since last import.")
```